apps/api/src/domain/domainModelValidators/__tests__/domainModelValidators/buildDomainModelValidatorTestCases/song.domainModelValidatorTestCase.ts

```diff
@@ -0,0 +1,90 @@
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { Song } from '../../../../models/song/song.entity';
+import { EntityId } from '../../../../types/ResourceId';
+import { resourceTypes } from '../../../../types/resourceTypes';
+import InvalidEntityDTOError from '../../../errors/InvalidEntityDTOError';
+import songValidator from '../../../songValidator';
+import { DomainModelValidatorTestCase } from '../../types/DomainModelValidatorTestCase';
+import getValidEntityInstaceForTest from '../utilities/getValidEntityInstaceForTest';
+
+const validDTO = getValidEntityInstaceForTest(resourceTypes.song).toDTO();
+
+const buildTopLevelError = (id: EntityId, innerErrors: InternalError[]) =>
+    new InvalidEntityDTOError(resourceTypes.song, id, innerErrors);
+
+export const buildSongTestCase = (): DomainModelValidatorTestCase<Song> => ({
+    resourceType: resourceTypes.song,
+    validator: songValidator,
+    validCases: [
+        {
+            dto: validDTO,
+        },
+    ],
+    invalidCases: [
+        {
+            description: 'audioURL is an empty string',
```

Comment:
Let's add a case for an invalidly formatted URL (but a non-empty string).

Reply:
404 Error?

Reply:
No. something like `myprotocol:wwww.soundcloud.com`

Reply:
resolved